src/artifactRegistry.ts

```diff
@@ -109,6 +183,13 @@ export class CachingDockerRegistryClient {
     },
   });
 
+  async getGitCommitsBetweenTags(
+    options: GitCommitsBetweenTagsOptions,
+  ): Promise<string[]> {
+    // For now we aren't caching anything since this will only run on promotion prs
```

Comment:
It is also running on dry runs (and the speed impact is notable) for better or for worse — we should talk about whether this is a good thing (more stuff getting tests on PRs) or not.

I think we should make the assumption that this call is relatively cacheable for the case where the tags start with `main---` since the next number is "number of commits in history of main" and we shouldn't be rewriting main.